scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/TerminationData.java

```diff
@@ -80,10 +80,38 @@
                 TaskResultImpl taskResult) {
             this.taskData = taskData;
             this.terminationStatus = terminationStatus;
+            taskResult.setPropagatedVariables(SerializationUtil.serializeVariableMap(restoreNonInheritedTaskVariables(taskResult)));
```

Comment:
The serialization could be embedded in the private method

Reply:
Done!